internal/project/service.go

```diff
@@ -182,3 +183,43 @@ func (s *ServiceImpl) GetProjectVersions(ctx context.Context, projectID int) ([]
 
 	return versions, nil
 }
+
+func (s *ServiceImpl) DeleteProjectVersionByIDAndProjectID(ctx context.Context, ID, projectID uint) error {
+	version, err := s.repo.GetVersionByIDAndProjectID(ctx, ID, projectID)
+	if err != nil {
+		if errors.Is(err, ErrNotFound) {
+			
+			return ErrNotFound
+		}
+
+		return errors.Wrapf(err, "Failed to retrieve project version with ID %d and project ID %d", ID, projectID)
+	}
+
+	return s.deleteProjectVersion(ctx, version)
+}
+
+func (s *ServiceImpl) deleteProjectVersion(ctx context.Context, version *Version) error {
+
+	deleteVersionByIDTransaction, err := s.repo.DeleteVersionByIDTransaction(ctx, version.ID)
```

Comment:
```suggestion
func (s *ServiceImpl) deleteProjectVersion(ctx context.Context, version *Version) error {
	deleteVersionByIDTransaction, err := s.repo.DeleteVersionByIDTransaction(ctx, version.ID)
```
Unnecessary whitespace